sklearn/model_selection/_search.py

```diff
@@ -204,6 +204,9 @@ class ParameterSampler:
     n_iter : integer
         Number of parameter settings that are produced.
 
+        .. versionchanged:: 0.20
+           `n-iter` now acts as an upper bound on iterations
+
```

Comment:
We can probably remove this one as it would mostly confuse readers I think.
If you want to keep it, I would suggest to say smth like "now throw a warning instead of raising an error when..."